tests/baselines/reference/assertionFunctionsCanNarrowByDiscriminant.js

```diff
@@ -30,5 +30,5 @@ var animal = { type: 'cat', canMeow: true };
 assertEqual(animal.type, 'cat');
 animal.canMeow; // since is cat, should not be an error
 var animalOrUndef = { type: 'cat', canMeow: true };
-assertEqual(animalOrUndef === null || animalOrUndef === void 0 ? void 0 : animalOrUndef.type, 'cat');
+assertEqual(animalOrUndef !== null && animalOrUndef !== void 0 && animalOrUndef.type, 'cat');

```

Comment:
This emit should not have changed. If the test were `assertEqual(animalOrUndef?.type, false)` then the test would result in a false negative.

Reply:
This has reverted after latest changes